syntax = "proto3";

import "google/protobuf/timestamp.proto";

package logger;

service Logger {
    rpc SaveMessage (LogMessageRequest) returns (LogMessageReply) {}

    rpc GetAllMessages (GetListMessageRequest) returns (ListLogMessagesReply) {}
}

message LogMessageRequest {
    string message = 1;
    google.protobuf.Timestamp created_at = 3
}

message GetListMessageRequest {}

message LogMessageReply {
    string result = 1;
}

message ListLogMessagesReply {
    repeated string messages = 1;
}